File: trabajo 18/pila.cs
using System;
using System.Collections.Generic;

class Program
{
    // Tamaño máximo de la pila
    const int MAX_SIZE = 100;

    // Lista para almacenar los elementos de la pila
    static List<int> stack = new List<int>();

    // Función para agregar un elemento a la pila
    static void Push(int item)
    {
        if (stack.Count == MAX_SIZE) // Verifica si la pila está llena
        {
            Console.WriteLine("Stack Overflow"); // Mensaje de error
            return;
        }
        stack.Add(item); // Agrega el elemento al final (parte superior de la pila)
    }

    // Función para eliminar y retornar el elemento superior de la pila
    static int Pop()
    {
        if (stack.Count == 0) // Verifica si la pila está vacía
        {
            Console.WriteLine("Stack Underflow"); // Mensaje de error
            return -1; // Retorna -1 para indicar que la pila está vacía
        }
        int top = stack[stack.Count - 1];
        stack.RemoveAt(stack.Count - 1); // Elimina el último elemento
        return top;
    }

    // Función para ver el elemento superior sin eliminarlo
    static int Peek()
    {
        if (stack.Count == 0)
        {
            Console.WriteLine("Pila vacía");
            return -1;
        }
        return stack[stack.Count - 1];
    }

    // Función para verificar si la pila está vacía
    static bool IsEmpty()
    {
        return stack.Count == 0;
    }

    // Función para verificar si la pila está llena
    static bool IsFull()
    {
        return stack.Count == MAX_SIZE;
    }

    // Programa principal
    static void Main()
    {
        Push(10); // Agrega elementos a la pila
        Push(20);
        Push(30);

        Console.WriteLine("Elemento Superior: " + Peek()); // Muestra el elemento superior
        Console.WriteLine("Extrae elemento: " + Pop()); // Elimina y muestra el elemento superior
        Console.WriteLine("Elemento Superior: " + Peek()); // Muestra el nuevo elemento superior
    }
}

File: trabajo 18/pila.js
// Tamaño máximo de la pila
const MAX_SIZE = 100;

// Arreglo para almacenar los elementos de la pila
let stack = [];

// Función para agregar un elemento a la pila
function push(item) {
    if (stack.length === MAX_SIZE) { // Verifica si la pila está llena
        console.log("Stack Overflow"); // Mensaje de error
        return;
    }
    stack.push(item); // Agrega el elemento al final (parte superior de la pila)
}

// Función para eliminar y retornar el elemento superior de la pila
function pop() {
    if (stack.length === 0) { // Verifica si la pila está vacía
        console.log("Stack Underflow"); // Mensaje de error
        return -1; // Retorna -1 para indicar que la pila está vacía
    }
    return stack.pop(); // Elimina y retorna el elemento superior
}

// Función para ver el elemento superior sin eliminarlo
function peek() {
    if (stack.length === 0) { // Verifica si la pila está vacía
        console.log("Pila vacía");
        return -1;
    }
    return stack[stack.length - 1]; // Retorna el elemento superior
}

// Función para verificar si la pila está vacía
function isEmpty() {
    return stack.length === 0;
}

// Función para verificar si la pila está llena
function isFull() {
    return stack.length === MAX_SIZE;
}

// Programa principal
function main() {
    push(10); // Agrega elementos a la pila
    push(20);
    push(30);

    console.log("Elemento Superior:", peek()); // Muestra el elemento superior
    console.log("Extrae elemento:", pop()); // Elimina y muestra el elemento superior
    console.log("Elemento Superior:", peek()); // Muestra el nuevo elemento superior
}

// Ejecutar el programa
main();
